secretmanager/api/SecretManagerTest/SecretManagerTest.cs

```diff
@@ -0,0 +1,160 @@
+// Copyright(c) 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+using System.IO;
+using Xunit;
+using System;
+using System.Linq;
+
+using Google.Api.Gax.ResourceNames;
+using Google.Cloud.SecretManager.V1Beta1;
+
+namespace GoogleCloudSamples
+{
+    /// <summary>
+    /// Runs the sample app's methods and tests the outputs
+    /// </summary>
+    public class SampleTests
+    {
+        private static readonly string s_projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+
+        readonly CommandLineRunner _secretManager = new CommandLineRunner()
+        {
+            VoidMain = SecretManagerSample.Main,
+            Command = "SecretManagerSample"
+        };
+
+        protected ConsoleOutput Run(params string[] args)
+        {
+            return _secretManager.Run(args);
+        }
+
+        public SampleTests()
+        {
+            if (s_projectId == null || s_projectId == "")
+            {
+                throw new Exception("missing GOOGLE_PROJECT_ID");
+            }
+        }
+
+        [Fact]
+        public void TestCreateAddAccessDelete()
```

Comment:
I'm a little worried about the maintenability of these tests that test multiple things. I understand that for testing delete you first need to create, etc. but I'd prefer a test that eclusively tests create, another delete, etc.
For doing that, consider creating a [Fixture](https://xunit.net/docs/shared-context#class-fixture) that can hold all the methods you need for setting up your tests and also for instance, a `void MarkForDeletion(Secret secret)` that saves a collection of secrets to delete, that are deleted on dispose. Take a look at this [StorageFixture](https://github.com/googleapis/google-cloud-dotnet/blob/master/apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/StorageFixture.cs) for instance. Let me know if I can help with that.

Reply:
A challenge with the other language samples was quotas. It depends on how fast the tests execute. Let me rewrite them and see if they aren't quota-ed out.

Reply:
A challenge with the other language samples was quotas. It depends on how fast the tests execute. Let me rewrite them and see if they aren't quota-ed out.

Reply:
Okay I just pushed up some commits that break the tests apart.